### PR TITLE
Experience/8962+8963/replace tables within message tracker

### PR DESCRIPTION
**Pull request by etanb**
This fix replaces the Message Search and the Message Tracker tables with our shared component.

Current state demo:

https://github.com/CDCgov/prime-reportstream/assets/516877/8046db1d-6b37-4dfb-b6fb-c7fa10287e88

Most importantly, we figured out a path forwards of how to use USWDS with CSS modules.

### CODE REVIEW

**Comment by jillian-hammer** [comment]
@etanb  yay looks great!

two notes:
- [nitpick, can be punted] can the scrollbar start BELOW the sticky header?
- the content of the cell looks like it's vertically center-aligned, can you adjust to being aligned to the top to make visual scanning easier?

**Comment by etanb** [comment]
> [nitpick, can be punted] can the scrollbar start BELOW the sticky header?

I'm going to go ahead and say this would introduce some complications as a "sticky" header is out of the flow of the document and exists on top of the content it's scrolling on top of. The only way we could potentially do this is add some empty / negative space at the top that would push the content down, but that might add a blank area within the scrolling we'd want to avoid.

If you want to make a ticket, I can make a mini "spike" into this, and see how possible this is within a reasonable timebox. cc @jillian-hammer

**Comment by jillian-hammer** [comment]
@etanb thank you! this looks good to me. I will hold off on any style comments as Tshering will provide those specs for you as she wraps #9460. however there is one inconsistency I'm noticing between tables that scroll horizontally & vertically versus vertically only.

To replicate:

**Horizontal & Vertical Scrolling**
Message ID `6cf1a59ad17b4f` first result:
image

**Vertical Scrolling only**
Message ID `d697b700-6b32-4bb6-954f-3dbb74f0cf99` first result:
image

**Notes**

- [ ] Vertical scrolling only table has extra white padding in the table headers (should be removed)
- [ ] Vertical scrolling table: the last column (currently `Transport Results`) is centered (should be left aligned)
- [ ] Adjust max height to 80vh (currently looks like 50vh but I may be using Inspect wrong)

**Comment by jillian-hammer** [comment]
> > [nitpick, can be punted] can the scrollbar start BELOW the sticky header?
> 
> I'm going to go ahead and say this would introduce some complications as a "sticky" header is out of the flow of the document and exists on top of the content it's scrolling on top of. The only way we could potentially do this is add some empty / negative space at the top that would push the content down, but that might add a blank area within the scrolling we'd want to avoid.
> 
> If you want to make a ticket, I can make a mini "spike" into this, and see how possible this is within a reasonable timebox. cc @jillian-hammer

all good! thanks for thinking about it – not a high priority. @etanb

File: frontend-react/src/uswds-loader.scss
Comment (jpandersen87) [inline frontend-react/src/uswds-loader.scss:1]: lets call this global-modules.scss

**Review by jpandersen87 — APPROVED** [review]
A comment on a file name, but otherwise g2g.

File: frontend-react/src/config/endpoints/messageTracker.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface RSMessageDetail {
 }
 
 export interface MessageListResource {
+    id?: number;
```
Comment (penny-lischer) [inline frontend-react/src/config/endpoints/messageTracker.ts:44]: nit, would this ever be null?

**Review by penny-lischer — APPROVED** [review]
LGTM

File: frontend-react/src/config/endpoints/messageTracker.ts (hunk 1 above)
Comment (etanb) [reply frontend-react/src/config/endpoints/messageTracker.ts]: This piece of code got around typing that: 

https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/components/Table/ColumnData.tsx#L80-L82

Comment (etanb) [reply frontend-react/src/config/endpoints/messageTracker.ts]: Going to update the MessageTrackerMockServer:

https://github.com/CDCgov/prime-reportstream/blob/fe8425f4b8897181204d47402133843c5738d688/frontend-react/src/__mocks__/MessageTrackerMockServer.ts#L28-L47

Let me know if you see any problems @penny-lischer

**Pull request merged**